### PR TITLE
Add xpath() method to the document returned by scrape module powered by lxml

### PR DESCRIPTION
**Pull request by gimite**

### CODE REVIEW

File: tests/scrape.py [hunk 1]
```diff
@@ -1030,6 +1034,46 @@ def enclosing(self, tagname=None, content=None, enders=[], **attrs):
         c = content is not None and ' with content %r' % content or ''
         raise ScrapeError('no %s found%s%s' % (tag, a, c))
 
+class Document(Region):
+    """A document returned as an HTTP response.
+    """
+
+    def __init__(self, content_str, headers, charset=None):
```
Comment (skywhale) [inline tests/scrape.py]: Please document that not passing charset will result in xpath() not working. I'd assume that the charset will be automatically determined by default.

Comment (skywhale) [inline tests/scrape.py]: Please document the behavior when charset=None

**Comment by skywhale** [comment]
LGTM. Thank you so much!

File: tests/scrape.py (hunk 1 above)
Comment (gimite) [inline tests/scrape.py]: Done. I also refactored the code to guess the charset from content-type header in Document.__init__ which was done in fetch().

**Comment by gimite** [comment]
Thanks. Merged.

**Pull request merged**